File: Go/101-symmetric-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
//检查两个树是不是对称
func isSymmetricTowTree(p, q *TreeNode) bool {
    if p == nil && q == nil {
        return true
    } else if p == nil || q == nil{
        return false
    } else {
        if p.Val != q.Val {
            return false
        }
        return isSymmetricTowTree(p.Left, q.Right) && isSymmetricTowTree(p.Right, q.Left)
    }
}


func isSymmetric(root *TreeNode) bool {
    if root == nil {
        return true
    }
    return isSymmetricTowTree(root.Left, root.Right)
}

File: Python/101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def isSymmetricTwoTree(p: TreeNode, q: TreeNode) -> bool:
            #都是空
            if p == None and q == None:
                return True
            elif p == None or q == None:
                #有一个为空
                return False
            else:
                #检查两个结点值, 以及左右子树是否对称
                return p.val == q.val and isSymmetricTwoTree(p.left, q.right) and isSymmetricTwoTree(p.right, q.left)
        
        if root == None:
            return True
        return isSymmetricTwoTree(root.left, root.right)